packages/kbn-elastic-agent-utils/src/agent_guards.ts

```diff
@@ -43,12 +54,21 @@ export function isRumOrMobileAgentName(agentName?: string) {
 }
 
 export function isIosAgentName(agentName?: string) {
-  return agentName?.toLowerCase() === 'ios/swift';
+  const lowercased = agentName && agentName.toLowerCase();
+
+  return (
+    lowercased?.startsWith('opentelemetry/swift') ||
+    IOS_AGENT_NAMES.includes(agentName! as IOSAgentName)
+  );
 }
 
 export function isAndroidAgentName(agentName?: string) {
   const lowercased = agentName && agentName.toLowerCase();
-  return lowercased === 'android/java';
+
+  return (
+    lowercased?.startsWith('opentelemetry/android') ||
```

Comment:
Q: Should this be an AND condition and not OR 🤔

Reply:
It should be an OR condition, as we the two conditions can not be true at the same time. Example: we can have 'android/java' but no 'opentelemetry/android', so it would fail for every agent name that is not OpenTelemetry

Reply:
Thanks for explaining I was thinking that we need both to be true, but it makes sense